lib/util/Components.js

```diff
@@ -373,22 +373,40 @@ function componentRule(rule, context) {
         return false;
       }
 
-      const returnsConditionalJSXConsequent = node[property] &&
+      const isReturnsConditionalJSX = () => {
+        const returnsConditionalJSXConsequent = node[property] &&
         node[property].type === 'ConditionalExpression' &&
         jsxUtil.isJSX(node[property].consequent);
-      const returnsConditionalJSXAlternate = node[property] &&
+        const returnsConditionalJSXAlternate = node[property] &&
         node[property].type === 'ConditionalExpression' &&
         jsxUtil.isJSX(node[property].alternate);
-      const returnsConditionalJSX = strict ?
-        (returnsConditionalJSXConsequent && returnsConditionalJSXAlternate) :
-        (returnsConditionalJSXConsequent || returnsConditionalJSXAlternate);
+        return strict ?
```

Comment:
I think `strict` property not used.
I should probably remove `strict` property from `isReturningJSX`.